### PR TITLE
update fms to fms/2022.04 and add a coupled mixed-mode test

### LINKED ISSUE #1508: Update FMS to fms/2022.04 and add a mixed-mode coupled test
## Description
The FMS library will be updated to fms/2022.04 for the ufs-weather-model. With this update, the fv3 dynamical core can be compiled with 32-bit while the MOM6 can be compiled with 64-bit in coupled applications. A new mixed-mode coupled test will also be added.

## Solution
Update modulefiles directory add a test in tests/tests directory.

## Alternatives
If applicable, add a description of any alternative solutions or features you've considered.

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/nems/issues/<issue_number>
- needed by noaa-emc/fv3atm/pull/<pr_number>

### PR DESCRIPTION
**Pull request by binli2337**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model.

## Description
1. Update FMS to FMS/2022.04
2. Add a new mixed-mode test

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!)
- fixes #1508 

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel
- [x] cheyenne.gnu
- [x] gaea.intel
- [x] jet.intel
- [x] wcoss2.intel
- [ ] acorn.intel
- [x] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/609
- waiting on https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/163
- waiting on https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/81

Co-authored-by @junwang-noaa

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@binli2337 I don't see the added test in the PR. Will it compare against the non-mixed mode baseline?

**Comment by binli2337** [comment]
@DeniseWorthen I just added the test. Thanks!

**Comment by DeniseWorthen** [comment]
Thanks @binli2337. So the baseline change is only because a test is being added. All other coupled tests do not use mixed-mode and should reproduce the current baselines, right?

Do we know that the mixed-mode test passes the ORTs?

**Comment by binli2337** [comment]
@DeniseWorthen You are correct. All other tests can reproduce existing baselines. The ORT tests passed a few months ago. The ORT tests have not been run for this PR.

**Comment by jkbk2004** [comment]
@binli2337 fms-2022.04 is ready on all RDHPCS. can you sync up the branch and resolve the conflict?

**Comment by binli2337** [comment]
@jkbk2004 I have a PR to update the fms_mixedmode branch at NOAA-EMC/GFDL_atmos_cubed_sphere.  After it is merged, I will update the fv3atm PR and the ufs-weather-model PR.

**Comment by jkbk2004** [comment]
@binli2337 sure! let me know. @zach1221 may combine in a minor change on jenkinsfile.

**Comment by binli2337** [comment]
@jkbk2004 The code is ready.

**Comment by DusanJovic-NOAA** [comment]
@binli2337 Did you test this branch on wcoss2?

**Comment by DusanJovic-NOAA** [comment]
Please change the title of this PR to mention "fms mixed mode". "update fv3 and add test" is not very descriptive.

**Comment by binli2337** [comment]
@DusanJovic-NOAA No. It was tested on Acorn a few weeks ago.

**Comment by jkbk2004** [comment]
@binli2337 new baseline, cpld_control_p8_mixedmode and no impact on all other baselines, right?

**Comment by DusanJovic-NOAA** [comment]
> @DusanJovic-NOAA No. It was tested on Acorn a few weeks ago.

I'm afraid if will fail on wcoss2. My branch which also uses fms/2022.04 failed to build yesterday on Cactus due to some issues with loading modules.
@aerorahul Can you please point Bin to your updated wcoss2 modulefile.

**Comment by binli2337** [comment]
@jkbk2004 Yes. Should I start the ORT tests now?

**Comment by jkbk2004** [comment]
@binli2337 Are you going to run ORT on hera? I will start creating new BL and testing RT from hera. Yes, let's start testing.

**Comment by binli2337** [comment]
@jkbk2004 Yes, I will run the ORT on Hera and update the log file.

**Comment by jkbk2004** [comment]
@binli2337 I like to check if we can combine in #1528 jenkinsfile change to this pr. @zach1221 your pr is nothing to do with baselines. Can we discuss about combining in your pr?

**Comment by zach1221** [comment]
@jkbk2004 Correct. #1528 will not change baselines.

**Comment by aerorahul** [comment]
> > @DusanJovic-NOAA No. It was tested on Acorn a few weeks ago.
> 
> I'm afraid if will fail on wcoss2. My branch which also uses fms/2022.04 failed to build yesterday on Cactus due to some issues with loading modules. @aerorahul Can you please point Bin to your updated wcoss2 modulefile.

I was able to build w/ FMS/2022.04 on Cactus yesterday by modifying the `ufs_wcoss2.intel.lua` modulefile.
The updates to the modulefile can be seen [here](https://github.com/ufs-community/ufs-weather-model/compare/develop...aerorahul:ufs-weather-model:feature/wcoss2-fms-2022.04?expand=1)
They are in a branch [feature/wcoss2-fms-2022.04](https://github.com/aerorahul/ufs-weather-model/tree/feature/wcoss2-fms-2022.04) on my fork.

Feel free to grab those changes.

**Comment by binli2337** [comment]
@jkbk2004 It is OK to merge this PR with #1528.  Should I do the merging now?

**Comment by binli2337** [comment]
@jkbk2004 ORT tests passed.

**Comment by jkbk2004** [comment]
> @jkbk2004 It is OK to merge this PR with #1528. Should I do the merging now?

Sure! please, go ahead to merge in #1528 if possible.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/1143179304/20221209190022/ufs-weather-model
Please make changes and add the following label back: orion-intel-RT

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA if wcoss2 is skipped, we can start merging process.

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA if wcoss2 is skipped, we can start merging process.

the atmaq tests failed, but it seems it may have been a machine issue. I'm running them again now. If successful i'll push in a little bit.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by binli2337** [comment]
@jkbk2004 I don't see the log on wcoss2 yet. Have the tests on wcoss2 passed?

**Comment by BrianCurtis-NOAA** [comment]
> @jkbk2004 I don't see the log on wcoss2 yet. Have the tests on wcoss2 passed?

I'm running 2 failed tests again. Should commit shortly if passed.

**Comment by binli2337** [comment]
@BrianCurtis-NOAA Thanks!

**Comment by jkbk2004** [comment]
@binli2337 we can merge in [cubed sphere pr 163](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/163). I am moving there to ask for the merge.

File: tests/ci/Jenkinsfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ pipeline {
                 axes {
                     axis {
                         name 'TEST_NAME'
-                        values 'control', 'regional_control', 'cpld_control_nowave_noaero_p8'
+                        values 'control_p8', 'cpld_control_nowave_noaero_p8'
```
Comment (junwang-noaa) [inline tests/ci/Jenkinsfile:16]: @jkbk2004 May I ask why we remove the "regional_control" test?

Comment (jkbk2004) [reply tests/ci/Jenkinsfile:16]: regional_control fails for a while now + fms update on-going + epic switched aws resources. Multiple possibilities so we like to turn it off util we get a fix. @zach1221 looking into the issues,

Comment (junwang-noaa) [reply tests/ci/Jenkinsfile:16]: Do we have an issue to track the problem? If not, would you or @zach1221 please create an issue? Thanks

Comment (jkbk2004) [reply tests/ci/Jenkinsfile:16]: sure! @zach1221 and I will follow up with creating issue.

**Comment by jkbk2004** [comment]
@binli2337 the fv3 dependency was merged. let's finish merging on this side.

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**